scripts/vendors/microsoft/configure.py

```diff
@@ -0,0 +1,37 @@
+import os
+import json
+import argparse
+
+def main(github_pool, num_lists):
+  github_runner = None
+  test_lists = []
+  platforms = ["windows-x64", "windows-arm64", "linux-x64", "linux-arm64", "macos-x64", "macos-arm64"]
+
+  for platform in platforms:
+    if platform in github_pool:
+      if "macos" in github_pool and "arm64" in github_pool:
+        github_runner = "config=['self-hosted', 'macOS', 'ARM64']"
+      elif "macos" in github_pool and "x64" in github_pool:
+        github_runner = "config='macos-11'"
+      else:
+        github_runner = f"config=['self-hosted', '1ES.Pool={ github_pool }']"
+
+  with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
+    print(github_runner, file=fh)
+
+  for list in range(num_lists):
+    test_lists.append(f"testList_{list}")
+
+  with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
+    print(f"testConfig={json.dumps(test_lists)}", file=fh)
```

Comment:
(putting all IO at the end)

```suggestion
  with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
    print(github_runner, file=fh)
    print(f"testConfig={json.dumps(test_lists)}", file=fh)
```

Reply:
...or better yet, have `main` call other functions to do non-IO work, returning the stuff you want to print out. Then, you can make far better testable code for later. (I know this is a trivial example, but it's a muscle building exercise).

Reply:
Exported non-io to their own functions. Condensed the file io statements into one.